libraries/chain/controller.cpp

```diff
@@ -174,8 +191,21 @@ struct controller_impl {
          db.undo();
       }
 
+      uint32_t unconf_blocknum = 1;
+      const auto& ubi = unconfirmed_blocks.get_index<unconfirmed_block_index,by_num>();
+      auto objitr = ubi.rbegin();
+      if( objitr != ubi.rend() ) {
+         unconf_blocknum = objitr->blocknum;
+      }
+
       FC_ASSERT( db.revision() == head->block_num, "fork database is inconsistent with shared memory",
-                 ("db",db.revision())("head",head->block_num) );
+                 ("db",db.revision())("head",head->block_num)("unconfimed",unconf_blocknum) );
+
+      edump((unconf_blocknum));
+      /*
+      FC_ASSERT( head->block_num == unconf_blocknum, "unconfirmed block database out of sync",
```

Comment:
If this assertion was failing, that is a sign that something is wrong with the code. Because as far as I see, the `unconf_blocknum` and `head->block_num` should never be out of sync.